Декоративные элементы всех страниц

tsbv opened the pull request:
Получилось пока так, не понимаю как сделать переносы строк в трёх блоках "Жильё" "Еда" "Сувениры", а точнее у их описаний. Тегов <br> будет слишком много. Ещё в блоке "Заинтересовались" тоже. 
Какие есть проблемы на данном этапе?
Иногда появляется чувство что вот-вот запутаюсь и всё сломается.
Как реализовать этот отступ у Главной страницы слева и справа и добавить тень?
Я примерно вижу так, что сейчас всё содержимое тега body нужно обернуть в div, и ему задавать стили? Какие?
Надеюсь скоординируете по ошибкам.

---
:mortar_board: [Декоративные элементы всех страниц](https://up.htmlacademy.ru/htmlcss/36/user/2277251/tasks/16)

:boom: https://htmlacademy-htmlcss.github.io/2277251-sedona-36/9/

tsbv committed:
Добавляет cursor:pointer для кнопки

tsbv committed:
Добавляет внешний отступ для секции

..для секции "Приезжайте в Седону. отдохнуть...,"

tsbv committed:
Добавляет левый отступ в форме

Для placeholder'a "Ваш e-mail"

tsbv committed:
Добавляет внешний отступ секции

..для секции "Седона - небольшой городок..."
Стало чуточку красивее:)

tsbv committed:
Удаляет ненужную обёртку

tsbv committed:
Поправляет фоновые цвета кнопка

tsbv committed:
Вставляет svg поиска и избранного

Я так понял, что пока с спанчиком с цифрой 12 пока не работает? пробую пробую и не получается пока!
Иконка Поиска и Избранного немного неровно стоят. Почему?

tsbv committed:
Добавляет подчёркивание пункта меню

А так же сделал по макету логотип, задав высоту хедеру, а z-index 1 логотипу.

tsbv committed:
Добавляет обертку списку в навигации

Прижимаю блок к правому краю.

tsbv committed:
Выравнивает размеры иконок в хедере

tsbv committed:
Выравнивает текст в блоках преимуществ по центру

tsbv committed:
Добавляет перенос строки по макету

tsbv committed:
добавляет размеры, отступы в блоках преимуществ.

tsbv committed:
Улучшает читаемость кода

(надеюсь:3)

tsbv committed:
Поправляет паддинги у кнопки в хедере

tsbv committed:
Вставляет свг-инлайн иконки

tsbv committed:
Добавляет декоративный элемент в блок преимуществ

у параграфа отключаю внешние отступы что сцентрировать декоративный элемент.

tsbv committed:
Поправляет код

tsbv committed:
Добавляет атрибуты тегу svg

Не забываем про атрибут aria-hidden="true" и focusable="false", чтобы скрыть иконку от скринридеров.

tsbv committed:
Добавляет divider в блок hero

Надеюсь добавил его правильно.
Так же добавил отступ для кнопки в хедере для сходства с макетом (что-то не совсем корректно сделано, с иконками не в ряд стоят.)

tsbv committed:
Исправляет размер элемента ссылки tel

tsbv committed:
добавляет стили для спан цифры

tsbv committed:
добавляет стили для спан цифры

tsbv committed:
выравнивает зелёный кружок с цифрой

tsbv committed:
Merge branch 'master' of https://github.com/tsa8a/2277251-sedona-36

tsbv committed:
Убирает лишние пробелы

tsbv committed:
Merge branch 'master' of https://github.com/tsa8a/2277251-sedona-36

tsbv committed:
Добавляет состояния элементов из стайлгайда

Вставил в разметку svg'шки  инлайново, убрал у них дополнительные значения fill, чтобы иконки стали красится при состояниях.
:focus почему то сработал только на кнопке, а на иконках не работает почему-то.

senchkim commented on index.html:
```diff
@@ -13,7 +13,7 @@
         </a>
```
https://pastenow.ru/19bc3c224f5c7d46304d78749e998e35
Я уже писал о том что изображения нужно называть с маленькой английской буквой и в качестве разделителя использовать тире.

Важно!
1) удаляешь ошибочные изображения и делаешь коммит.
2) добавляешь уже исправленные изображения из шага-1 вновь и делаешь коммит.

senchkim commented on styles/styles.css:
```diff
@@ -126,16 +167,19 @@ body {
 .hero {
   width: 1200px;
   height: 485px;
+  background-image: url("../images/divider-main.svg");
```
Почему тут в строке передаешь путь до изображения, а на строке 159 нет?
Надо сделать одинаково!

senchkim commented on styles/styles.css:
```diff
@@ -126,16 +167,19 @@ body {
+  background-repeat: no-repeat;
+  background-position: bottom;
 }
 
```
https://pastenow.ru/e5fbbdddaa4336257383553db1ea4554
Зачем дублировать множество раз одни и те же стили? 
Создай класс `.reset-list` - задай его всем тэгам `ul` и потом 1 раз обнули:
```
.reset-list {
  padding: 0;
  margin: 0;
  list-style-type: none;
}
```

senchkim commented on styles/styles.css:
```diff
@@ -264,13 +346,15 @@ body {
 
 .advantages-bridge-wrapper {
-  text-align: center;
+  width: 400px;
```
Зачем добавляешь уже же есть?

senchkim commented on styles/styles.css:
```diff
@@ -264,13 +346,15 @@ body {
 .advantages-bridge-wrapper {
-  text-align: center;
+  width: 400px;
+  height: 385px;
```
Зачем добавляешь уже же есть?

senchkim commented:
Привет)
1) Отступы - сделать их просто используя `<br />` / либо по играйся через `min-width`
2) Что касается тени - то сделать ее просто создай класс типа `page-container` - оберни весь сайт сразу после `<body>` - и задай в стилях тень `.page-container {
  box-shadow: 0px 0px 15px rgba(0, 0, 0, 0.2);
}`

senchkim commented on index.html:
```diff
@@ -22,26 +22,56 @@
               <span class="visually-hidden">
                 Поиск
               </span>
             </a>
```
ты используешь вот такую картинку `sedona-because-the-grand-canyon-sucks.svg` - хотя ее нет в папки `images` ((

senchkim commented:
Резюмируя:
В целом вопросов нет - ты молодец проделал огромную работу)

1) Нужно упростить код - я описал уже в комментариях
2) Нужно доработать стайлгайд - ховер/фокус/актив
3) ну и по мелочи иконки подвинуть, где-то отступы поправить)

tsbv commented:
У меня в папке всё с маленькой буквы, не знаю почему у вас так отображается, странно. Щас попробую что нибудь сделать
 
 
Георгий Артурович Ц.
 
  
>Пятница, 3 марта 2023, 13:52 +03:00 от Dimetrius ***@***.***>:
> 
> 
>@senchkim commented on this pull request.
>----------------------------------------------------------------------
>In  index.html :
>> @@ -13,7 +13,7 @@
>         </a>
>
>https://pastenow.ru/19bc3c224f5c7d46304d78749e998e35
>Я уже писал о том что изображения нужно называть с маленькой английской буквой и в качестве разделителя использовать тире.
>Важно!
>*  удаляешь ошибочные изображения и делаешь коммит.
>*  добавляешь уже исправленные изображения из шага-1 вновь и делаешь коммит.
>----------------------------------------------------------------------
>In  styles/styles.css :
>> @@ -126,16 +167,19 @@ body {
> .hero {
>   width: 1200px;
>   height: 485px;
>+  background-image: url("../images/divider-main.svg");
>
>Почему тут в строке передаешь путь до изображения, а на строке 159 нет?
>Надо сделать одинаково!
>----------------------------------------------------------------------
>In  styles/styles.css :
>> @@ -126,16 +167,19 @@ body {
> .hero {
>   width: 1200px;
>   height: 485px;
>+  background-image: url("../images/divider-main.svg");
>+  background-repeat: no-repeat;
>+  background-position: bottom;
> }
> 
>
>https://pastenow.ru/e5fbbdddaa4336257383553db1ea4554
>Зачем дублировать множество раз одни и те же стили?
>Создай класс .reset-list - задай его всем тэгам ul и потом 1 раз обнули:
>.reset-list {
>  padding: 0;
>  margin: 0;
>  list-style-type: none;
>}
>
>----------------------------------------------------------------------
>In  styles/styles.css :
>> @@ -264,13 +346,15 @@ body {
> }
> 
> .advantages-bridge-wrapper {
>-  text-align: center;
>+  width: 400px;
>
>Зачем добавляешь уже же есть?
>----------------------------------------------------------------------
>In  styles/styles.css :
>> @@ -264,13 +346,15 @@ body {
> }
> 
> .advantages-bridge-wrapper {
>-  text-align: center;
>+  width: 400px;
>+  height: 385px;
>
>Зачем добавляешь уже же есть?
>—
>Reply to this email directly,  view it on GitHub , or  unsubscribe .
>You are receiving this because you authored the thread. Message ID:  <htmlacademy-htmlcss/2277251-sedona-36/pull/9/review/1323515303 @ github . com>
 

tsbv committed:
Удаляет ошибочные изображения

tsbv committed:
Добавляет исправленные изображения

tsbv committed:
Исправляет все пути до изображений

tsbv committed:
Добавляет класс reset-list

Удаляю лишние классы
Обнуляю у тегов <h3> и <p> дополнительно отступы, иначе блок съезжает.

tsbv committed:
Добавляет всем тегам спискам класс reset-list

tsbv committed:
Удаляет лишние теги

tsbv committed:
Удаляет лишние теги и пробелы

tsbv committed:
Улучшает читаемость кода

tsbv committed:
Добавляет состояние кнопкам

ховер/фокус/актив

tsbv committed:
Добавляет состояния кнопке в рассылке

ховер/фокус/актив

tsbv committed:
Удаляет лишний тег

tsbv committed:
Исправляет переносы строк

tsbv committed:
Добавляет тень главной странице

tsbv committed:
Удаляет лишний код

tsbv committed:
уменьшает размер блока

Для соответствия с макетом, не используя тг <br>.
Текст "Там есть мост дьявола" теперь выглядит как надо.

tsbv committed:
Добавляет исправленные svg иконки

tsbv committed:
Добавляет другой вариант переноса строки

tsbv committed:
Исправляет отступы в хедере

tsbv committed:
Добавляет фавиконку главной странице

tsbv committed:
Удаляет лишний код

tsbv committed:
Поправляет отступы в шапке

tsbv committed:
Исправляет отступы у цифры 12 в шапке

tsbv committed:
Удаляет лишние классы

tsbv committed:
Добавляет состояния ссылкам в навигации шапки

TEST-07. Стайлгайд.
Если в стайлгайде не предусмотрены состояния элемента, то их нужно реализовать на своё усмотрение.

tsbv committed:
Удаляет тег href у главного логотипа

tsbv committed:
Добавляет комментарии в стилях

tsbv committed:
Удаляет лишнюю обёртку

tsbv committed:
Добавляет перенос строки для текста

tsbv committed:
Добавляет шрифт и отступы в блоке с формой

tsbv committed:
Добавляет файл favicon.ico в корень проекта

Критерий HTML-08. Фавиконки.
Подключена только иконка размером 32×32 в формате PNG. favicon.ico не подключён к странице, но лежит в корне проекта.
